### PR TITLE
enable pylint:useless-return

### PR DESCRIPTION
**Pull request by natmokval**
Issue #48855. This PR enables pylint type "R" warning:  "useless-return". 
In one place (https://github.com/natmokval/pandas/blob/da3025ae6dec3fed73df0cf1d1bbc5ef0d816695/pandas/tests/groupby/test_apply.py#L151) the warning is disabled to match the style of surrounding functions. 

- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/tests/groupby/test_apply.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def f_scalar(group):
         names.append(group.name)
         return 0
 
-    def f_none(group):
+    def f_none(group):  # pylint: disable= "useless-return"
         # GH10519, GH12155, GH21417
         names.append(group.name)
         return None
```
Comment (MarcoGorelli) [inline pandas/tests/groupby/test_apply.py:154]: can't this one be removed as well?

Comment (natmokval) [reply pandas/tests/groupby/test_apply.py:154]: Functions above and below this function have the same functionality and differ only by return type. To make the code more readable I left `return None`.
 
Should I remove it too?

Comment (MarcoGorelli) [reply pandas/tests/groupby/test_apply.py:154]: yeah I think that's better

Comment (natmokval) [reply pandas/tests/groupby/test_apply.py]: Thanks. I removed it too and added fixup commit to the PR.

**Comment by MarcoGorelli** [comment]
Thanks @natmokval - I think the CI failure's unrelated but I've merged upstream/main to restart the checks to be sure

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Sweet, thanks @natmokval

**Comment by natmokval** [comment]
Thank you @mroeschke

**Pull request merged**